prow/git/localgit/localgit.go

```diff
@@ -32,13 +32,25 @@ import (
 
 type Clients func() (*LocalGit, v2.ClientFactory, error)
 
+func DefaultBranch(dir string) string {
```

Comment:
Why scatter invocations of this all over the place into package-level vars rather than just hardcoding any value for `InitialBranch` and exposing it via a `const`?

Reply:
Because the default branch is not a constant

Reply:
It varies per system, per git repo, per version of git.

Reply:
I am aware, that is why `localgit` should always explicitly set it (aka default `InitialBranch` to the value of an exported `const` and use that `const` rather than the current `var`)

Reply:
nah

Reply:
I do not want to depend on the `git init --initial-branch=foo` flag being available, as it was only added in July 2020:

* https://git-scm.com/docs/git-init/2.27.0
* https://git-scm.com/docs/git-init/2.28.0

Reply:
But you do already by having added an option for that. Besides, why not? Developers tend to have fairly current versions of git and CI obviously works as well

Reply:
I didn't though. The flag is only used on one test and only when the default branch is not master, which is only possible when an old version of git is used